doc/source/whatsnew/v0.24.0.txt

```diff
@@ -513,7 +513,7 @@ Missing
 MultiIndex
 ^^^^^^^^^^
 
--
+- Removed legacy internal formats (:issue:`21654`)
```

Comment:
Putting it here again: the above message is unclear to me what it is about. 

Either this change is not relevant for the user (eg only internal clean-up), and not whatsnew notice is needed. Or it is, but then needs to be clarified (eg do we drop support for very old pickle files?)

Reply:
if that is the case, I would maybe just remove the what's new notice? (since the 0.13 threshold is already documented I assume?)

Reply:
yeah I don't think this has any user visible effects, but still having a whatsnew note just documents what we did. I would expand the note slightly but leave it.

Reply:
We don't always add a whatsnew note for internal code clean-up (if it has no user visible change), so I still don't think it is needed here. Anyhow, if we keep one, it should be comprehensible for users (the current one is not even comprehensible for me without looking at the issue)